src/main/java/edu/cornell/kfs/module/cg/businessobject/lookup/CuAwardLookupableHelperServiceImpl.java

```diff
@@ -0,0 +1,131 @@
+package edu.cornell.kfs.module.cg.businessobject.lookup;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.lang3.StringUtils;
+import org.kuali.kfs.kns.lookup.HtmlData;
+import org.kuali.kfs.kns.lookup.HtmlData.AnchorHtmlData;
+import org.kuali.kfs.kns.service.DocumentHelperService;
+import org.kuali.kfs.kns.web.struts.form.LookupForm;
+import org.kuali.kfs.kns.web.ui.Column;
+import org.kuali.kfs.kns.web.ui.ResultRow;
+import org.kuali.kfs.krad.util.GlobalVariables;
+import org.kuali.kfs.krad.util.KRADConstants;
+import org.kuali.kfs.module.ar.ArConstants;
+import org.kuali.kfs.module.cg.CGConstants;
+import org.kuali.kfs.module.cg.businessobject.lookup.AwardLookupableHelperServiceImpl;
+import org.kuali.kfs.sys.KFSConstants;
+import org.kuali.kfs.sys.KFSPropertyConstants;
+import org.kuali.rice.kim.api.KimConstants;
+import org.kuali.rice.kim.api.permission.PermissionService;
+import org.kuali.rice.krad.bo.BusinessObject;
+
+import edu.cornell.kfs.sys.CUKFSConstants;
+
+@SuppressWarnings("deprecation")
+public class CuAwardLookupableHelperServiceImpl extends AwardLookupableHelperServiceImpl {
+    private static final long serialVersionUID = 1372707190594999024L;
+    private static final org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(CuAwardLookupableHelperServiceImpl.class);
+    
+    private static final String VIEW_INVOICES_LINK_VALUE = "View Invoices";
+    private static final String INVOICE_LINK_COLUMN_NAME = "invoiceLink";
+    
+    protected DocumentHelperService documentHelperService;
+    protected PermissionService permissionService;
+    
+    @Override
+    public List<HtmlData> getCustomActionUrls(BusinessObject businessObject, List pkNames) {
+        LOG.debug("getCustomActionUrls, entering");
+        List<HtmlData> anchorHtmlDataList = new ArrayList<HtmlData>();
+        
+        if (canInitAward()) {
+            anchorHtmlDataList.add(getUrlData(businessObject, KRADConstants.MAINTENANCE_EDIT_METHOD_TO_CALL, pkNames));
+            if (allowsMaintenanceNewOrCopyAction()) {
+                anchorHtmlDataList.add(getUrlData(businessObject, KRADConstants.MAINTENANCE_COPY_METHOD_TO_CALL, pkNames));
+            }
+        }
+
+        if (canViewInvoiceLink() && getAccountsReceivableModuleBillingService().isContractsGrantsBillingEnhancementActive()) {
+            AnchorHtmlData invoiceUrl = getInvoicesLookupUrl(businessObject);
+            anchorHtmlDataList.add(invoiceUrl);
+        }
+
+        return anchorHtmlDataList;
+    }
+    
+    @Override
+    public List<Column> getColumns() {
+        List<Column> columns = super.getColumns();
+
+        if (canInitAward() || !canViewInvoiceLink()) {
```

Comment:
Should the invoice link column also get removed when the CGB enhancement is inactive?

Reply:
I don't think so, but will confirm

Reply:
yes, I do need to add this check, thanks for noticing